pennylane/devices/tests/test_compare_default_qubit.py

```diff
@@ -0,0 +1,211 @@
+            and dev.capabilities()["supports_tensor_observables"]
+        )
+
+        if not supports_tensor:
+            pytest.skip("Device does not support tensor observables.")
+
+        if not dev.analytic:
+            pytest.skip("Device is in non-analytical mode.")
+
+        theta = 0.432
+        phi = 0.123
+
+        def circuit(theta, phi):
+            qml.RX(theta, wires=[0])
+            qml.RX(phi, wires=[1])
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(wires=0) @ qml.PauliZ(wires=1))
+
+        qnode_def = qml.QNode(circuit, dev_def)
+        qnode = qml.QNode(circuit, dev)
+
+        grad_def = qml.grad(qnode_def, argnum=[0, 1])
+        grad = qml.grad(qnode, argnum=[0, 1])
+
+        assert np.allclose(qnode(theta, phi), qnode_def(theta, phi), atol=tol(dev.analytic))
+        assert np.allclose(grad(theta, phi), grad_def(theta, phi), atol=tol(dev.analytic))
+
+    @pytest.mark.parametrize("ret", [qml.expval, qml.var])
+    def test_random_circuit(self, device, tol, ret):
+        """Test that the random expectation value is correct"""
+        n_wires = 2
+        dev = device(n_wires)
+        dev_def = qml.device("default.qubit", wires=n_wires)
+
+        if dev.name == dev_def.name:
+            pytest.skip("Device is default.qubit.")
+
+        supports_tensor = (
+            "supports_tensor_observables" in dev.capabilities()
+            and dev.capabilities()["supports_tensor_observables"]
+        )
+
+        if not supports_tensor:
+            pytest.skip("Device does not support tensor observables.")
+
+        if not dev.analytic:
+            pytest.skip("Device is in non-analytical mode.")
+
+        n_layers = np.random.randint(1, 5)
+        weights = 2 * np.pi * np.random.rand(n_layers, 1)
+
+        def circuit(weights):
+            RandomLayers(weights, wires=range(n_wires))
+            return ret(qml.PauliZ(wires=0) @ qml.PauliX(wires=1))
+
+        qnode_def = qml.QNode(circuit, dev_def)
+        qnode = qml.QNode(circuit, dev)
+
+        grad_def = qml.grad(qnode_def, argnum=0)
+        grad = qml.grad(qnode, argnum=0)
+
+        assert np.allclose(qnode(weights), qnode_def(weights), atol=tol(dev.analytic))
+        assert np.allclose(grad(weights), grad_def(weights), atol=tol(dev.analytic))
+
+    def test_four_qubit_random_circuit(self, device, tol):
+        """Test a four-qubit random circuit with the whole set of possible gates"""
+        n_wires = 4
+        dev = device(n_wires)
+        dev_def = qml.device("default.qubit", wires=n_wires)
+
+        if dev.name == dev_def.name:
+            pytest.skip("Device is default.qubit.")
+
+        supports_tensor = (
+            "supports_tensor_observables" in dev.capabilities()
+            and dev.capabilities()["supports_tensor_observables"]
+        )
+
+        if not supports_tensor:
+            pytest.skip("Device does not support tensor observables.")
```

Comment:
```suggestion
```
Only to be accepted if https://github.com/PennyLaneAI/pennylane/pull/897/files#r537898582 is accepted.

I am suggesting this because I think the other tests do a good job of looking at tensor products, and by restricting to tensor products we may be excluding some devices. So let's just do `qml.expval(qml.PauliZ(0))` and try and cover more devices.